test/langtools/tools/javac/flags/FlagsTest.java

```diff
@@ -23,28 +24,98 @@
+
+    public static void main(String[] args) throws Throwable {
+        verifyFlagsNonOverlapping();
+        findFreeFlags();
+    }
+
+    private static void verifyFlagsNonOverlapping() throws Throwable {
+        Map<FlagTarget, Map<Long, List<Field>>> target2Flag2Fields = computeTarget2Flag2Fields();
+
+        for (Entry<FlagTarget, Map<Long, List<Field>>> targetAndFlag : target2Flag2Fields.entrySet()) {
+            for (Entry<Long, List<Field>> flagAndFields : targetAndFlag.getValue().entrySet()) {
+                if (flagAndFields.getValue().size() > 1) {
+                    throw new AssertionError("duplicate flag for target: " + targetAndFlag.getKey() + ", flag: " + flagAndFields.getKey() + ", flags fields: " + flagAndFields.getValue());
+                }
+            }
+        }
+    }
+
+    private static void findFreeFlags() throws Throwable {
+        Map<FlagTarget, Map<Long, List<Field>>> target2Flag2Fields = computeTarget2Flag2Fields();
+
+        long freeTypeFlags = ~collectFlags(target2Flag2Fields, FlagTarget.TYPE);
+        long freeMethodFlags = ~collectFlags(target2Flag2Fields, FlagTarget.METHOD);
+        long freeVariableFlags = ~collectFlags(target2Flag2Fields, FlagTarget.VARIABLE);
+
+        printFreeFlags("TYPE", freeTypeFlags);
+        printFreeFlags("METHOD", freeMethodFlags);
+        printFreeFlags("VARIABLE", freeVariableFlags);
+    }
+
+    private static Map<FlagTarget, Map<Long, List<Field>>> computeTarget2Flag2Fields() throws IllegalArgumentException, IllegalAccessException, AssertionError {
+        Map<FlagTarget, Map<Long, List<Field>>> target2Flag2Fields = new HashMap<>();
         for (Field f : Flags.class.getFields()) {
-            if (!Modifier.isStatic(f.getModifiers())) {
+            if (f.isAnnotationPresent(NotFlag.class)) {
                 continue;
             }
-            long flag = ((Number) f.get(null)).longValue();
-            try {
-                Flags.asFlagSet(flag);
-            } catch (AssertionError e) {
-                throw new AssertionError("missing Flags enum constant for: " + f.getName(), e);
+
+            Use use = f.getAnnotation(Use.class);
+
+            if (use == null) {
+                throw new AssertionError("No @Use and no @NotFlag for: " + f.getName());
+            }
+
+            long flagValue = ((Number) f.get(null)).longValue();
+
+            for (FlagTarget target : use.value()) {
+                target2Flag2Fields.computeIfAbsent(target, _ -> new HashMap<>())
+                        .computeIfAbsent(flagValue, _ -> new ArrayList<>())
+                        .add(f);
             }
         }
+        return target2Flag2Fields;
+    }
+
+    private static void printFreeFlags(String comment, long freeFlags) {
+            System.err.print("free flags for " + comment + ": ");
+            for (int bit = 16; bit < 64; bit++) { //lowest 16 bits are used in classfiles, never suggest adding anything there
```

Comment:
Replace 16 with Character.SIZE and 64 with Long.SIZE?

Reply:
I don't think `Character.SIZE` is really appropriate, but I've declared `U2_SIZE` for `16`, and used `Long.SIZE` for `64`.